programs/ssl/ssl_server2.c

```diff
@@ -2732,6 +2741,14 @@ int main( int argc, char *argv[] )
     if( opt.cert_req_ca_list != DFL_CERT_REQ_CA_LIST )
         mbedtls_ssl_conf_cert_req_ca_list( &conf, opt.cert_req_ca_list );
 
+#if defined(MBEDTLS_X509_CRT_PARSE_C)
+#if defined(MBEDTLS_KEY_EXCHANGE_CERT_REQ_ALLOWED_ENABLED)
+    /* exercise setting DN hints for server certificate request */
```

Comment:
Perhaps add a comment that using the server cert is not something we're likely to want to do in real life and we do that only for testing purposes? This bugged me a bit initially, though it does make sense once I thought about it. (Same for `set_hs` below.)

Reply:
My reasoning is that it's unlikely for the same certificate to be used both to sign other certificates, and as the server certificate. So, as you say, in practice you'd probably use an intermediate CA's certificate to configure DN hints, but IMO that would most likely be distinct from the certificate used to authenticate your server during the handshake. Not a problem for testing, but IMO worth mentioning that real-world use would differ (since `ssl_server2` has a kind of dual-purpose as a test tool, but also an example that users can look at).

Reply:
Anyway, that's also something we can adjust later when we address guards and using `set_hs` from a callback, so I'm marking this conversation as resolved.